### PR TITLE
feat: support query data by use cypher language

### PR DESCRIPTION
**Pull request by jadepeng**
#1748

### CODE REVIEW

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/gremlin/CypherAPI.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        TranslationFacade translator = new TranslationFacade();
+        String gremlin = translator.toGremlinGroovy(cypher);
+        gremlin = this.buildQueryableGremlin(graph, gremlin);
+        return gremlin;
+    }
+
+    private String buildQueryableGremlin(String graph, String gremlin) {
+        gremlin = "g = " + graph + ".traversal()\n" + gremlin;
+
+        // hg not support single
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/gremlin/CypherAPI.java:84]: improve to add more details

**Review by javeme — COMMENTED** [review]
Nice feature! Thanks for your contribution

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/gremlin/CypherAPI.java (hunk 1 above)
Comment (jadepeng) [reply hugegraph-api/src/main/java/com/baidu/hugegraph/api/gremlin/CypherAPI.java]: Already added more note and test case

**Review by javeme — DISMISSED** [review]

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/gremlin/CypherAPI.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.HttpHeaders;
+import jakarta.ws.rs.core.Response;
+
+@Path("graphs/{graph}/cypher")
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/gremlin/CypherAPI.java:23]: could you please also bump up api version and adapt java-client:
https://github.com/apache/incubator-hugegraph/blob/master/hugegraph-api/src/main/java/com/baidu/hugegraph/version/ApiVersion.java#L124

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/gremlin/CypherAPI.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        TranslationFacade translator = new TranslationFacade();
+        String gremlin = translator.toGremlinGroovy(cypher);
+        gremlin = this.buildQueryableGremlin(graph, gremlin);
+        return gremlin;
+    }
+
+    private String buildQueryableGremlin(String graph, String gremlin) {
+        /*
+         * `CREATE (a:person { name : 'test', age: 20) return a`
+         * would be translated to :
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/gremlin/CypherAPI.java:84]: "translated to :" => "translated to:"

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/gremlin/CypherAPI.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+         * `g.addV('person').as('a').property(single, 'name', 'test') ...`,
+         * but hugegraph don't support `.property(single, k, v)`,
+         * so we replace it to .property(k, v) here
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/gremlin/CypherAPI.java:87]: \`.property(k, v)`

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/gremlin/CypherAPI.java (hunk 2 above)
Comment (jadepeng) [reply hugegraph-api/src/main/java/com/baidu/hugegraph/api/gremlin/CypherAPI.java:23]: see https://github.com/apache/incubator-hugegraph-toolchain/pull/272

File: hugegraph-api/src/main/java/com/baidu/hugegraph/version/ApiVersion.java [hunk 5]
```diff
@@ -118,10 +118,11 @@ public final class ApiVersion {
      * [0.66] Issue-1567: Support get schema RESTful API
      * [0.67] Issue-1065: Support dynamically add/remove graph
      * [0.68] Issue-1763: Support adamic-adar & resource-allocation API
+     * [0.69] Issue-1748: Support Cypher query RESTful API
      */
 
     // The second parameter of Version.of() is for IDE running without JAR
-    public static final Version VERSION = Version.of(ApiVersion.class, "0.68");
+    public static final Version VERSION = Version.of(ApiVersion.class, "0.69");
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/version/ApiVersion.java:125]: need to update implement-version: https://github.com/apache/incubator-hugegraph/commit/69ac3cc07279178e0f5c1059123d3e1709298f20#diff-3744a042d40e54198e87c7271da06363141165f2523541c71e81c820486ca87fR156

Comment (jadepeng) [reply hugegraph-api/src/main/java/com/baidu/hugegraph/version/ApiVersion.java:125]: updated

**Review by javeme — APPROVED** [review]

**Review by coderzc — APPROVED** [review]
LGTM

**Review by imbajin — APPROVED** [review]
we could add more cases in client side

**Pull request merged**